File: frontend/src/lib/shared/types/Verdict.ts
// Verdict response from the backend API
// Matches Go's VerdictResponse structure
export interface Verdict {
    /** Whether the submitted item was identified as furniture */
    admissible: boolean;
    /** Straightness score from 1-10 (10 being perfectly straight) */
    score: number;
    /** Detailed verdict components */
    verdict: VerdictDetails;
    /** Unique request identifier */
    requestId: string;
    /** ISO 8601 timestamp of the verdict */
    timestamp: string;
}

// Detailed verdict components
// Matches Go's VerdictDetails structure
export interface VerdictDetails {
    /** The furniture offense (e.g., "Scheefhangende Zitting") */
    crime: string;
    /** The punishment or acknowledgment */
    sentence: string;
    /** Legal justification with fictitious law articles */
    reasoning: string;
    /** What the judge observed in the photo */
    observation: string;
    /** The verdict classification */
    verdictType: "vrijspraak" | "waarschuwing" | "schuldig";
}

File: backend/internal/core/domain/verdict.go
package domain

// VerdictResponse represents the full verdict response from the API
type VerdictResponse struct {
	Admissible bool           `json:"admissible"`
	Score      int            `json:"score"`
	Verdict    VerdictDetails `json:"verdict"`
	RequestID  string         `json:"requestId"`
	Timestamp  string         `json:"timestamp"`
}

// VerdictDetails contains the structured components of the legal verdict
type VerdictDetails struct {
	Crime     string `json:"crime"`     // The furniture offense
	Sentence  string `json:"sentence"`  // The punishment
	Reasoning string `json:"reasoning"` // Legal justification
}

// PhotoMetadata contains information about an uploaded photo
type PhotoMetadata struct {
	Filename    string `json:"filename"`
	ContentType string `json:"contentType"`
	Size        int64  `json:"size"`
}
